daemon/src/setup_contract.rs

```diff
@@ -568,20 +580,28 @@ pub async fn roll_over(
                         })?;
 
                     let maker_amount = tx
-                        .find_output_amount(&maker_script_pubkey)
+                        .find_output_amount(&maker_address.script_pubkey())
                         .unwrap_or_default();
                     let taker_amount = tx
-                        .find_output_amount(&taker_script_pubkey)
+                        .find_output_amount(&taker_address.script_pubkey())
                         .unwrap_or_default();
-
-                    Ok(Cet {
+                    let cet = Cet {
                         maker_amount,
                         taker_amount,
                         adaptor_sig: *other_encsig,
                         range: digits.range(),
                         n_bits: digits.len(),
                         txid: tx.txid(),
-                    })
+                    };
+
+                    debug_assert_eq!(
```

Comment:
I think this is fair, but it would be interesting to have some stats because rollovers are already expensive and we would do that for *each* CET. 
Could you maybe put that into a separate PR and we merge it later and try to get some stats on it on `testnet`?

Reply:
The change only applies to debug builds! So it doesn't affect production. I can put it in a separate PR, but I would merge it straight away 😛